x/oracle/handler.go

```diff
@@ -21,6 +22,42 @@ func NewHandler(keeper Keeper) sdk.Handler {
 
 // Handle a message to make a bridge claim
 func handleMsgMakeBridgeClaim(ctx sdk.Context, keeper Keeper, msg MsgMakeBridgeClaim) sdk.Result {
-	//do it
-	return sdk.Result{} // return
+	//check if prophecy exists or not
+	//if exist
+	//	//get it and continue checks
+	//	//check if claim for this validator exists or not
+	//	//if does
+	//		//return error
+	//	//else
+	//		//add claim to list
+	//	//check if claimthreshold is passed
+	//	//if does
+	//		//check enough claims match and are valid
+	//		//update prophecy to be successful
+	//		//trigger minting
+	//		//save finalized prophecy to db
+	//		//return
+	//	//if doesnt
+	//		//save updated prophecy to db
+	//		//return
+	//else (if doesnt exist yet)
+	bridgeClaim := NewBridgeClaim(msg.Nonce, msg.EthereumSender, msg.CosmosReceiver, msg.Validator, msg.Amount)
+	bridgeClaims := []BridgeClaim{bridgeClaim}
+	newProphecy := NewBridgeProphecy(msg.Nonce, PendingStatus, getMinimumClaims(), bridgeClaims)
+	err := keeper.createProphecy(ctx, newProphecy)
+	if err != nil {
+		return err.Result()
+	}
+	return sdk.Result{}
+}
+
+func getMinimumClaims() int {
+	minimumClaims := float64(getTotalNumberValidators()) * DefaultConsensusNeeded
```

Comment:
`minimumClaims` should be the minimum power needed, not the minimum number of validators.